godot-core/src/classes/type_safe_replacements.rs

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+//! Replaces existing Godot APIs with more type-safe ones where appropriate.
+//!
+//! Each entry here *must* be accompanied by
+//!
+//! See also sister module [super::manual_extensions].
+
+use crate::builtin::{Callable, GString, StringName, Variant};
+use crate::classes::notify::NodeNotification;
+use crate::classes::object::ConnectFlags;
+use crate::classes::scene_tree::GroupCallFlags;
+use crate::classes::{Object, SceneTree, Script};
+use crate::global::Error;
+use crate::meta::{arg_into_ref, AsArg, ToGodot};
+use crate::obj::{EngineBitfield, Gd};
+
+impl Object {
+    pub fn get_script(&self) -> Option<Gd<Script>> {
+        let variant = self.raw_get_script();
+        if variant.is_nil() {
+            None
+        } else {
+            Some(variant.to())
+        }
+    }
+
+    pub fn set_script(&mut self, script: impl AsArg<Option<Gd<Script>>>) {
+        arg_into_ref!(script);
+
+        self.raw_set_script(&script.to_variant());
+    }
+
+    pub fn connect(&mut self, signal: impl AsArg<StringName>, callable: &Callable) -> Error {
+        self.raw_connect(signal, callable)
+    }
+
+    pub fn connect_flags(
```

Comment:
`connect_flags` is confusing – enum ConnectFlags (duh) is named the same and it looks like getter due to naming convention https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter. Maybe "connect_with_flags`? A bit mouthful tho